kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/container/orchestration/ImageConfiguration.java

```diff
@@ -0,0 +1,145 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Eurotech and/or its affiliates and others
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Eurotech
+ *******************************************************************************/
+
+package org.eclipse.kura.container.orchestration;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * Object which represents a image configuration used to request the generation
+ * of a new image instance
+ *
+ * @noimplement This interface is not intended to be implemented by clients.
+ * @since 2.4
+ *
+ */
+@ProviderType
+public class ImageConfiguration {
+    
+    private String imageName;
+    private String imageTag;
+    private Optional<RegistryCredentials> registryCredentials;
+    private int imageDownloadTimeoutSeconds = 500;
+    
+    private ImageConfiguration() {
+    }
+
+    /***
+     * Returns a Image's name as a String.
+     * 
+     * @return
+     */
+    public String getImageName() {
+        return this.imageName;
+    }
+
+    /***
+     * Returns a Image's tag as a String.
+     * 
+     * @return
+     */
+    public String getImageTag() {
+        return this.imageTag;
+    }
+
+    /***
+     * Returns a Image's download timeout time as a int.
+     * 
+     * @return
+     */
+    public int getimageDownloadTimeoutSeconds() {
+        return this.imageDownloadTimeoutSeconds;
+    }
+
+    /**
+     * Returns the Registry credentials
+     *
+     * @return
+     */
+    public Optional<RegistryCredentials> getRegistryCredentials() {
+        return this.registryCredentials;
+    }
+
+    /**
+     * Creates a builder for creating a new {@link ImageConfiguration} instance.
+     *
+     * @return the builder.
+     */
+    public static ImageConfigurationBuilder builder() {
+        return new ImageConfigurationBuilder();
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(this.imageName, this.imageTag, this.registryCredentials);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof ImageConfiguration)) {
+            return false;
+        }
+        ImageConfiguration other = (ImageConfiguration) obj;
+        return Objects.equals(this.imageName, other.imageName) && Objects.equals(this.imageTag, other.imageTag)
+                && Objects.equals(this.registryCredentials, other.registryCredentials);
+    }
+
+    public static final class ImageConfigurationBuilder {
+
+        private String imageName;
+        private String imageTag;
```

Comment:
Can we initialise this by default as latest? So that we don't have to deal with nulls in the rest of the code if the user does not explicitly initialise the value in the builder?

Reply:
Otherwise, we may use optional also here